### PR TITLE
Fix #580 & #581

### LINKED ISSUE #580: Can't exit product edition
Happens when coming from ingredient analysis, but possibly in other scenarii.
Steps to reproduce in this video: 
https://photos.app.goo.gl/VArjpock2VEJ3riQ9

### PR DESCRIPTION
**Pull request by philippeauriach**

### CODE REVIEW

**Comment by teolemon** [comment]
What if I scan another product quickly and the signal fires after that ?

File: Sources/Models/DataManager.swift [hunk 1]
```diff
@@ -295,6 +299,9 @@ class DataManager: DataManagerProtocol {
         productApi.postProduct(product, rawParameters: nil, onSuccess: {
             DispatchQueue.main.async {
                 onSuccess()
+                if let barcode = product.barcode {
```
Comment (philippeauriach) [inline Sources/Models/DataManager.swift]: @teolemon the view is reloaded for the current product, and only if the notification concerns the current displayed product

File: Sources/ViewControllers/Products/Detail/ProductDetailViewController.swift [hunk 2]
```diff
@@ -45,6 +45,16 @@ class ProductDetailViewController: ButtonBarPagerTabStripViewController, DataMan
...
         }
         setUserAgent()
+
+
+        NotificationCenter.default.addObserver(forName: .productChangesUploaded, object: nil, queue: .main) { [weak self] notif in
+            guard let barcode = notif.userInfo?["barcode"] as? String else {
+                return
+            }
+            if barcode == self?.product.barcode {
+                self?.refreshProduct {}
```
Comment (philippeauriach) [inline Sources/ViewControllers/Products/Detail/ProductDetailViewController.swift]: @teolemon the current displayed product is reloaded only if the update concerns this product

Comment (teolemon) [reply Sources/ViewControllers/Products/Detail/ProductDetailViewController.swift]: 👍

**Comment by teolemon** [comment]
lgtm. @aleene can you merge and release ?

**Pull request merged**